contracts/api3-server-v1/proxies/Api3CrossRateReaderProxyV1.sol

```diff
@@ -0,0 +1,193 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.27;
+
+import "../../vendor/@openzeppelin/contracts@5.0.2/proxy/utils/UUPSUpgradeable.sol";
+import "../../vendor/@openzeppelin/contracts-upgradeable@5.0.2/access/OwnableUpgradeable.sol";
+import "../../vendor/@chainlink/contracts@1.2.0/src/v0.8/shared/interfaces/AggregatorV2V3Interface.sol";
+import "./interfaces/IApi3CrossRateReaderProxyV1.sol";
+import "../../interfaces/IApi3ReaderProxy.sol";
+
+/// @title UUPS-upgradeable IApi3CrossRateReaderProxyV1 and
+/// AggregatorV2V3Interface implementation that is designed to be deployed by
+/// Api3CrossRateReaderProxyV1Factory
+/// @notice The owner of this contract is allowed to upgrade it. In the case
+/// that it is deployed by Api3CrossRateReaderProxyV1Factory, the owner will be
+/// the owner of Api3CrossRateReaderProxyV1Factory at the time of deployment.
+/// @dev For a gas-cheap `read()` implementation, this upgradeable contract
+/// uses immutable variables (rather than initializable ones). To enable this,
+/// an Api3CrossRateReaderProxyV1 needs to be deployed for each unique
+/// combination of variables. The end user does not need to concern themselves
+/// with this, as Api3CrossRateReaderProxyV1Factory abstracts this detail away.
+/// Refer to https://github.com/api3dao/migrate-from-chainlink-to-api3 for more
+/// information about the Chainlink interface implementation.
+contract Api3CrossRateReaderProxyV1 is
+    UUPSUpgradeable,
+    OwnableUpgradeable,
+    AggregatorV2V3Interface,
+    IApi3CrossRateReaderProxyV1
+{
+    /// @notice First IApi3ReaderProxy contract address
+    address public immutable override proxy1;
```

Comment:
Note that there is also a case when there is a non Api3ReaderProxy feed that we need to compose with. I think we should require `AggregatorV2V3Interface` support, but it would be nice to have a contract that allows wrapping a feed reader supporting `AggregatorV2V3Interface` that scales it to 1e18 so that we can support such case using  this wrapper.

Reply:
Then I would just use `latestAnswer()` and not expect the `read()` interface

Reply:
I have only found cases like [this](https://seitrace.com/address/0x3e10e6eF3c602885a9Cafd9785BD2143c52fbFF1?chain=pacific-1&contract=read_contract&slug=&tab=contract) where one of the proxies is not an `IApi3ReaderProxy` (or legacy `IProxy`) and they add a wrapped that implements a `read()` method with the same [signature](https://seitrace.com/address/0xaFb041c4268Efbf217ACEea6D795FA16A3b31A2D?chain=pacific-1&tab=contract) as the one in `IApi3ReaderProxy` interface.

Reply:
I think I might have found a case where they use a wrapper to convert the value to 8 decimals https://fraxscan.com/address/0x68E1085A7639e80257f7f159A611Dd944278A24e#code#F12#L46

Reply:
https://github.com/api3dao/contracts/pull/472/commits/012d779b03e408bde97ef75e23a558c698deab1d

Reply:
We can now wrap `AggregatorV2V3Interface` with [NormalizedApi3ReaderProxyV1](https://github.com/api3dao/contracts/pull/472/commits/2fe44e816ce20ff0775324bd86b71f2b510d25c7)